CMakeLists.txt

```diff
@@ -51,58 +66,87 @@ if(APPLE)
+
+# builtin `list(TRANSFORM)` (https://cmake.org/cmake/help/latest/command/list.html#transform)
+# was only added in cmake 3.12, but we target 3.10, so we define a custom
+# functions which can prepend a prefix to all list elements.
+# copied from https://stackoverflow.com/a/27630120
+function(PREPEND var prefix)
+	set(listVar "")
+	foreach(f ${ARGN})
+		list(APPEND listVar "${prefix}/${f}")
+	endforeach(f)
+	set(${var} "${listVar}" PARENT_SCOPE)
+endfunction()
+
+# ===== sentry library =====
+
+add_library(sentry "${PROJECT_SOURCE_DIR}/vendor/mpack.c")
+add_library(sentry::sentry ALIAS sentry)
+add_subdirectory(src)
+set_target_properties(sentry PROPERTIES PUBLIC_HEADER "include/sentry.h")
+
+# https://gitlab.kitware.com/cmake/cmake/issues/18393
 if(BUILD_SHARED_LIBS)
-	add_definitions(-DSENTRY_BUILD_SHARED)
-else()
-	add_definitions(-DSENTRY_BUILD_STATIC)
+	if(APPLE)
+		sentry_install(FILES "$<TARGET_FILE:sentry>.dSYM" DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+	elseif(MSVC)
+		sentry_install(FILES "$<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:$<TARGET_PDB_FILE:sentry>>"
+			DESTINATION "${CMAKE_INSTALL_BINDIR}")
+	endif()
 endif()
 
-FIND_PACKAGE(CURL)
-if(CURL_FOUND)
-	set(WITH_CURL TRUE)
-	add_definitions(-DSENTRY_WITH_LIBCURL_TRANSPORT)
-	include_directories(${CURL_INCLUDE_DIR})
-	set(LINK_LIBRARIES ${LINK_LIBRARIES} ${CURL_LIBRARIES})
+if(BUILD_SHARED_LIBS)
+	target_compile_definitions(sentry PRIVATE SENTRY_BUILD_SHARED)
+else()
+	target_compile_definitions(sentry PRIVATE SENTRY_BUILD_STATIC)
 endif()
-if(WIN32)
-	add_definitions(-DSENTRY_WITH_WINHTTP_TRANSPORT)
-	set(LINK_LIBRARIES ${LINK_LIBRARIES} "winhttp.lib")
+
+if(SENTRY_CURL_SUPPORT)
+	find_package(CURL REQUIRED)
+	target_compile_definitions(sentry PUBLIC SENTRY_WITH_LIBCURL_TRANSPORT)
+	target_include_directories(sentry PRIVATE ${CURL_INCLUDE_DIR})
+	target_link_libraries(sentry PRIVATE $<BUILD_INTERFACE:${CURL_LIBRARIES}>)
 endif()
 
-if(BACKEND_CRASHPAD)
-	set(CMAKE_CXX_STANDARD 14)
-else()
-	set(CMAKE_CXX_STANDARD 11)
+if(WIN32)
+	target_compile_definitions(sentry PUBLIC SENTRY_WITH_WINHTTP_TRANSPORT)
+	target_link_libraries(sentry PRIVATE winhttp)
 endif()
 
 #set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror=implicit-function-declaration -Werror=incompatible-function-pointer-types -Wall -fvisibility=hidden")
-if(NOT WIN32)
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -fvisibility=hidden")
+if(MSVC)
+	if(CMAKE_SIZEOF_VOID_P EQUAL 4)
+		set(CMAKE_ASM_MASM_FLAGS "${CMAKE_ASM_MASM_FLAGS} /safeseh")
+	endif()
+else()
+	#set_property(TARGET sentry PROPERTY C_VISIBILITY_PRESET hidden)
+	target_compile_options(sentry PUBLIC $<BUILD_INTERFACE:-Wall>)
 endif()
 
-if(BACKEND_CRASHPAD AND WIN32)
+if(BACKEND_CRASHPAD AND MSVC)
 	# Disable duplicate-define warnings, as the crashpad build defines all the
 	# "slim windows.h" defines, which sentry_boot also defines.
-	add_definitions(/wd4005)
+	target_compile_options(sentry PUBLIC $<BUILD_INTERFACE:/wd4005>)
```

Comment:
since you explicitly check for these with `#if !defined` now in code, maybe remove this from the cmake file.

Reply:
Yeah, I did not check what was the reason for adding this warning disabler option.

Reply:
I could remove them.